Extract "webhooks & events" to separate page

orzechdev opened the pull request:
It introduces the following:
- extracted custom apps with webhook and events out of the rest external apps,
- moved the list of custom apps and details views from `/apps/` paths to `/custom-apps/` and refactored routing.

Closes https://github.com/saleor/saleor-app-marketplace/issues/133

<!-- Please mention all relevant issue numbers. -->

**PR intended to be tested with API branch:** <!-- For example: feature/warehouses  -->

### Screenshots

<!-- If your changes affect the UI, providing "before" and "after" screenshots will
greatly reduce the amount of work needed to review your work. -->
<img width="1087" alt="Screenshot 2022-12-13 at 16 26 31" src="https://user-images.githubusercontent.com/9825562/207375095-7beb427b-0f8a-4150-9df2-8aaeb3dd3283.png">

### Pull Request Checklist

<!-- Please keep this section. It will make maintainer's life easier. -->

1. [ ] This code contains UI changes
2. [ ] All visible strings are translated with proper context including data-formatting
3. [ ] Attributes `[data-test-id]` are added for new elements
4. [ ] Changes are mentioned in the changelog
5. [ ] The changes are tested in different browsers and in light/dark mode

### Test environment config

<!-- Do not remove this section. It is required to properly setup test deployment instance.
Modify API_URI if you want test instance to use custom backend. CYPRESS_API_URI is optional, use when necessary. -->

API_URI=https://automation-dashboard.staging.saleor.cloud/graphql/
MARKETPLACE_URL=https://apps.saleor.io
SALEOR_APPS_ENDPOINT=https://apps.saleor.io/api/saleor-apps

### Do you want to run more stable tests?
To run all tests, just select the stable checkbox. To speed up tests, increase the number of containers. Tests will be re-run only when the "run e2e" label is added.

1. [ ] stable
2. [ ] giftCard
3. [ ] category
4. [ ] collection
5. [ ] attribute
6. [ ] productType
7. [ ] shipping
8. [ ] customer
9. [ ] permissions
10. [ ] menuNavigation
11. [ ] pages
12. [ ] sales
13. [ ] vouchers
14. [ ] homePage
15. [ ] login
16. [ ] orders
17. [ ] products
18. [x] app
19. [ ] plugins
20. [ ] translations
21. [ ] navigation
22. [ ] variants
23. [ ] payments

CONTAINERS=1

Droniu commented on src/custom-apps/index.tsx:
```diff
@@ -0,0 +1,122 @@
+import CustomAppWebhookDetailsView from "./views/CustomAppWebhookDetails";
+
+const CustomAppList: React.FC<RouteComponentProps> = () => {
+  const qs = parseQs(location.search.substr(1));
```
It's a good idea, but I believe we should do refactor it everywhere at once. I've made an issue for this: #2836

andrzejewsky replied:
fair point, we want to migrate from `qa` anyway

lkostrowski replied:
Ok, so lets leave it like it is in this PR @orzechdev

Droniu commented on src/custom-apps/views/CustomAppList.tsx:
```diff
@@ -0,0 +1,127 @@
+
+const getCurrentAppName = (
+  id: string,
+  collection?: AppsListQuery["apps"]["edges"],
```
**suggestion:** We probably have a fragment for that, something like AppFragment.

orzechdev replied:
I have extracted it into `findById` util.

Droniu commented on src/custom-apps/views/CustomAppList.tsx:
```diff
@@ -0,0 +1,127 @@
+        name={getCurrentAppName(params.id, customApps)}
+        onClose={closeModal}
+        onConfirm={handleRemoveConfirm}
+        type={"CUSTOM"}
```
```suggestion
        type="CUSTOM"
```

andrzejewsky commented on src/custom-apps/urls.ts:
```diff
@@ -0,0 +1,57 @@
+) =>
+  customAppWebhookAddPath(encodeURIComponent(appId)) +
+  "?" +
+  stringifyQs(params);
```
+1 we need to do similar thing with our config

andrzejewsky reviewed (changes requested):
Please write unit tests (it's part of our DoD), at least for the components/code you have created. Please think also about testing the existing code you amended (at least critical path)

lkostrowski commented:
@andrzejewsky AFAIK this PR doesn't have new features, it extracted part of the apps page to a separate page. I agree to write unit tests (as always) but let's not block Marketplace release by writing tests to legacy code that already worked before. I believe proper testing will require a huge refactor which can be a separate PR (to be safe not mix tests with functionality)

andrzejewsky commented:
@lkostrowski this is part of our legacy management, writing units even for the legacy code is very important for us. Yeah, it worked, but we have no guarantee it will continue to work. It's not really recommended way to skip tests and treat them as a blocker, it's rather part of the implementation. I'm asking for simple units just critical stuff, I don't expect 100% coverage.

If that's really urgent, let's skip it. However, I really ask for including writing tests in the estimations (at least simple ones)

lkostrowski commented:
> @lkostrowski this is part of our legacy management, writing units even for the legacy code is very important for us. Yeah, it worked, but we have no guarantee it will continue to work. It's not really recommended way to skip tests and treat them as a blocker, it's rather part of the implementation. I'm asking for simple units, for just critical stuff, I don't expect 100% coverage.
> 
> If that's really urgent, let's skip it. However, I really ask for including writing tests in the estimations (at least simple ones)

I agree, I'm just not sure if unit tests are the best

From my experience legacy code is easier to test with higher-level tests (observable behavior, integrations), while units are best when code is refactored/extracted. 

I'm just worried that code is not testable and we will spend days on mocking or refactoring to write units :)

So I only ask to be pragmatic

orzechdev commented:
@lkostrowski @Droniu @andrzejewsky 

I have added:
- objects with resolvers for custom apps urls,
- `findById` util,
- some unit tests for utils

Updated:
- icon for "Webhooks & Events" card on configuration page

lkostrowski reviewed (approved):
LGTM

In next PRs I think we should unify naming (local apps / webhooks & events)

szczecha commented:
@orzechdev  @karolm-saleor  
Please make sure before approving and merging this ticket that e2e tests (apps.js) will work. Right now 3 of them fail probably due to visiting the new URL. 
<img width="447" alt="image" src="https://user-images.githubusercontent.com/30683248/207667023-85c80925-baf7-451c-a2c6-e648fec056c3.png">
<img width="738" alt="Screenshot 2022-12-14 at 18 35 35" src="https://user-images.githubusercontent.com/30683248/207667067-f0c68f94-a213-41ae-a057-1cfde8024f4b.png">

karolm-saleor commented:
✓ QA Passed
